src/components/AppContext/AppContext.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useCallback, useMemo, useState } from 'react';
+
+import { Todo } from '../../types/Todo';
+import { FilterMode } from '../../types/FilterMode';
+
+interface IContextValue {
```

Comment:
Consider using a more descriptive name for the interface, like `AppContextValue` instead of `IContextValue`.